src/SFA.DAS.EmployerAccounts/Commands/CreateLegalEntity/CreateLegalEntityCommandHandler.cs

```diff
@@ -100,21 +100,17 @@ public async Task<CreateLegalEntityCommandResponse> Handle(CreateLegalEntityComm
             var agreementView = await _accountRepository.CreateLegalEntityWithAgreement(createParams);
 
             agreementView.HashedAgreementId = _hashingService.HashValue(agreementView.Id);
-
-            await CreateAuditEntries(owner, agreementView);
-
-            await NotifyLegalEntityCreated(message.HashedAccountId, agreementView.LegalEntityId);
-
             var accountId = _hashingService.DecodeValue(message.HashedAccountId);
-
-            await EvaluateEmployerLegalEntityAgreementStatus(owner.AccountId, agreementView.LegalEntityId);
-
             agreementView.AccountLegalEntityPublicHashedId = _accountLegalEntityPublicHashingService.HashValue(agreementView.AccountLegalEntityId);
 
-            await PublishLegalEntityAddedMessage(accountId, agreementView.Id, createParams.Name, owner.FullName(), agreementView.LegalEntityId,
-                agreementView.AccountLegalEntityId, agreementView.AccountLegalEntityPublicHashedId, message.Code, message.Address, message.Source, ownerExternalUserId);
-
-            await PublishAgreementCreatedMessage(accountId, agreementView.Id, createParams.Name, owner.FullName(), agreementView.LegalEntityId, ownerExternalUserId);
+            await Task.WhenAll(
+                CreateAuditEntries(owner, agreementView),
```

Comment:
Same as before if we can have the tasks rather than each item, but I guess thinking about it, its more personal preference for readability, I know the `EmployerTeamController` TaskWhen all is the bane of my existence on occasion so don't know if this syntax gives a better experience?